src/components/toaster/toaster.tsx

```diff
@@ -0,0 +1,26 @@
+import * as RadixToast from '@radix-ui/react-toast';
+import { Toast } from './toast/toast';
+import { useToast } from './toaster-context';
+import classNames from 'classnames';
+
+import styles from './toaster.module.scss';
+
+export function Toaster() {
+    const { toastData: toast, onOpenChange, closeToast } = useToast();
+
+    if (!toast) return null;
+
+    return (
+        <RadixToast.Provider>
```

Comment:
I guess it's better if provider and viewport are always rendered, not conditionally.